server/src/test/java/org/elasticsearch/cluster/metadata/DataStreamTests.java

```diff
@@ -332,6 +332,96 @@ public void testRemoveBackingWriteIndex() {
         );
     }
 
+    public void testRemoveFailureStoreIndex() {
+        long currentTimeMillis = System.currentTimeMillis();
+        int numBackingIndices = randomIntBetween(2, 32);
+        int numFailureIndices = randomIntBetween(2, 32);
+        int indexToRemove = randomIntBetween(1, numFailureIndices - 1);
+        String dataStreamName = randomAlphaOfLength(10).toLowerCase(Locale.ROOT);
+
+        List<Index> indices = new ArrayList<>(numBackingIndices);
+        for (int k = 1; k <= numBackingIndices; k++) {
+            indices.add(new Index(DataStream.getDefaultBackingIndexName(dataStreamName, k), UUIDs.randomBase64UUID(random())));
+        }
+        List<Index> failureIndices = new ArrayList<>(numFailureIndices);
+        for (int k = 1; k <= numFailureIndices; k++) {
+            failureIndices.add(
+                new Index(DataStream.getDefaultFailureStoreName(dataStreamName, k, currentTimeMillis), UUIDs.randomBase64UUID(random()))
+            );
+        }
+        DataStream original = DataStreamTestHelper.newInstance(dataStreamName, indices, failureIndices);
```

Comment:
Should we extract the creation of the data stream in a helper method? With a quick look I think most of it is reusable.